mne_bids/path.py

```diff
@@ -52,11 +54,12 @@ class BIDSPath(object):
         The recording name for this item. Corresponds to "rec".
     space : str | None
         The coordinate space for an anatomical file. Corresponds to "space".
-    prefix : str | None
-        The prefix for the filename to be created. E.g., a path to the folder
-        in which you wish to create a file with this name.
-    suffix : str | None
-        The suffix for the filename to be created. E.g., 'audio.wav'.
+    root : str | None
```

Comment:
Currently, I think this can operate as bids_root, or just the parent directory. Should we just have it be explicitly `bids_root` because otherwise, people should use `make_bids_folders` to create/work with their directory paths?

This then brings up the question... should BIDSPath actually be `BIDSFPath` since it's not a representation of any path (i.e. it wouldn't be used for directories).

What do people think? @hoechenberger @jasmainak

Reply:
> Currently, I think this can operate as bids_root, or just the parent directory. Should we just have it be explicitly `bids_root` because otherwise, people should use `make_bids_folders` to create/work with their directory paths?

Not sure if I fully understand your question or the problem, but yes, this should be the BIDS root.



> This then brings up the question... should BIDSPath actually be `BIDSFPath` since it's not a representation of any path (i.e. it wouldn't be used for directories).

`BIDSFile` then? :) but `File` would imply that it can be opened and read… while this here is actually just the **name** of a file. `BIDSFilePath`? Seems long and awkward… `BIDSFilename`? mmh. Not convinced! But if you do want to change it, I'd be in favor of `BIDSFilename`, I guess…

Reply:
> > Currently, I think this can operate as bids_root, or just the parent directory. Should we just have it be explicitly `bids_root` because otherwise, people should use `make_bids_folders` to create/work with their directory paths?
> 
> Not sure if I fully understand your question or the problem, but yes, this should be the BIDS root.

Should `root` be renamed `bids_root` to be consistent with the other API? Is there also a way to check if the `root` is the actual bids root vs say like a sub-directory?

> > This then brings up the question... should BIDSPath actually be `BIDSFPath` since it's not a representation of any path (i.e. it wouldn't be used for directories).
> 
> `BIDSFile` then? :) but `File` would imply that it can be opened and read… while this here is actually just the **name** of a file. `BIDSFilePath`? Seems long and awkward… `BIDSFilename`? mmh. Not convinced! But if you do want to change it, I'd be in favor of `BIDSFilename`, I guess…

Heh if no one cares, idc unless it becomes a common "mental mistake" users get.

Reply:
> Should `root` be renamed `bids_root` to be consistent with the other API?

I would say rename it to `bids_root` for now and let's change it to `root` in all places in a followup PR?